### PR TITLE
Practice Testing pull request for the TSE.

### PR DESCRIPTION
**Pull request by RoodzFernando**
# Functions we wrote
* capitalize
* reverseString
* calculator
*  caesar cipher
* array analysis
# Tests implemented
* capitalize
* reverseString
* calculator
*  caesar cipher
* array analysis

### CODE REVIEW

File: functions.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    reverse: (str) => {
+        return str.split("").reverse().join("");
+    },
+    calculator: {
+        add: (x, y) => x + y,
+        subtract: (x, y) => x - y,
+        divide: (x, y) => x / y,
+        multiply: (x, y) => x * y,
+    },
```
Comment (bellom) [inline functions.js:13]: opps, you test should handle positive and negative scenario.
if I introduce a bug on the calculation methods test still passes
example: `line 9` I changed `x+y` to `x*y`, to my surprise test still passes.

File: functions.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+const ourTest = {
```
Comment (bellom) [inline functions.js:1]: kindly practice good naming conversion `ourTest` could be something more professional.

File: functions.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            'y', 'z'
+        ];
+        let output = "";
+        str.split("").forEach(letter => {
+            const letterIndex = letters.indexOf(letter);
+            if ((letterIndex + num) > 25) {
+                output += letters[(letterIndex + num) % 25];
+            } else {
+                output += letters[letterIndex + num];
+            }
```
Comment (bellom) [inline functions.js:29]: we still need to test for the following;
1. same case
2. punctuation!
image

File: functions.js (hunk 1 above)
Comment (MarvellousUbani) [reply functions.js:13]: Hi Bello,

Thanks for your feedback, for this particular test I think we were just to test for the presence of the methods inside the object and not necessarily the calculations

Comment (bellom) [reply functions.js:13]: Hi @MarvellousUbani, 

Yes, you might think like that but this project basically teaching us how to do testing in jest right and I think one of the core reasons why we write tests is to mitigate the time spend on debugging, so we can easily fix the methods that failed on our test result.

According to Odin instructions, you are to test all functions;
image
https://www.theodinproject.com/courses/javascript/lessons/testing-practice

I hope this helps!
Happy Coding 👍

**Pull request merged**